sublime_linter.py

```diff
@@ -153,6 +153,18 @@ def on_activated_async(self, view):
             return
 
         if has_syntax_changed(view):
+            # Stop here if we already have errors for this view. Some linters
+            # can report different errors for a file depending on whether it
+            # was linted explicitly or through another file. If we would lint
+            # again, the errors could be cleared in such a case. Instead we
+            # broadcast an event to give everyone a chance to update the new
+            # view.
+            if util.get_filename(view) in persist.file_errors:
+                events.broadcast('loaded_file_with_errors', {
+                    'filename': util.get_filename(view)
+                })
+                return
+
             hit(view, "on_load")
```

Comment:
This has edge-cases. Why is relinting such a file a bad idea? Is it generally a bad idea, or just for some linters?

Reply:
Yes okay, but the linter will relint maybe in background, on next save etc. and will eventually produce confusing results. So to solve the c++ thing we would probably a different approach. (I actually thought c++ linters would skip linting *just* header files.)

Anyway, I don't like the conditional here. If the user really just changed the syntax, we *want* to relint. It's probably okay to have explicit state here. If we don't find a `view` in `group_by_filename_and_update` for some filename, we store that information in e.g. `awaiting_view = Set[FileName]`. We check that here, flip the switch, draw and return. 

Ideally though, I hoped that this would be local to `highlights_view.py` in that in receives updates 'on_lint_result' just like before and remembers for which filename it couldn't yet draw the highlights. And I think that probably the c++ linter should reject linting on load in that particular case.